Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2144 @@
+    """
+    This function creates a default payload based on the demisto.args().
+    Args:
+        args: demisto.args()
+        p_limit: Page Limit (int)
+    Returns:
+        Return arguments dict.
+    """
+    # Default payload
+    payload: Dict[str, Any] = {'verb': 'query'}
+    page_size = args.get('page_size')
+    max_limit = args.get('limit', DEFAULT_PAGE_SIZE)
+    if page_size: 
+        if p_limit >= max_limit:
+        # in case user pass the page_size or limit is less than page_size
+            payload['limit'] = p_limit
+    else:
+        payload['limit'] = p_limit if p_limit != DEFAULT_PAGE_SIZE else max_limit
+
+    from_time = iso_creation_date(args.get('start_time',''))
+    to_time = iso_creation_date(args.get('end_time',''))
+
+    if from_time:
+        payload['start_time'] = from_time
+
+    if to_time:
+        payload['end_time'] = to_time
+
+    return payload
+
+
+def get_params_query(args: Dict, p_limit: int = 0) -> Dict:
+    """
+    This function creates a query param based on the demisto.args().
+    Args:
+        args: demisto.args()
+        p_limit: Page Limit (int)
+    Returns:
+        Return arguments dict.
+    """
+    query_param: Dict = {}
+    ip = args.get('ip')
+    url = args.get('url')
+    md5 = args.get('md5')
+    sha256 = args.get('sha256')
+
+    # if ip := args.get('ip'):
+    if ip:
+        check_valid_indicator_value('ip', ip)
+
+    # if url:
+    #     check_valid_indicator_value('urls', url)
+
+    if md5:
+        check_valid_indicator_value('md5', md5)
+
+    if sha256:
+        check_valid_indicator_value('sha256', sha256)
+
+    max_limit = args.get('limit', DEFAULT_PAGE_SIZE)
+    page_size = args.get('page_size')
+    if page_size and (p_limit > max_limit):
+        # in case user pass the page_size or limit, limit will be ignore
+        query_param['limit'] = p_limit
+    else:
+        query_param['limit'] = p_limit if p_limit != DEFAULT_PAGE_SIZE else max_limit
+
+    query_param['ip'] = ip
+    query_param['url'] = url
+    query_param['sha256'] = sha256
+    # query_param['incident_trigger_sig_id'] = args.get('incident_trigger_sig_id') # Parameter is Deprecated
+    query_param['id'] = arg_to_number(args.get('allowlist_id'), arg_name='allowlist_id')
+    query_param['domain'] = args.get('domain')
+
+    return query_param
+
+
+def check_valid_indicator_value(indicator_type: str,indicator_value: str) -> bool:
+    """
+    Check the validity of indicator values
+    Args:
+        indicator_type: Indicator type provided in the command
+            Possible Indicator type are : sha256, urls, domain, ip, md5
+        indicator_value: Indicator value provided in the command
+    Returns:
+        True if the provided indicator values are valid
+    """
+    if indicator_type == 'sha256':
+        if not re.match(sha256Regex, indicator_value):
+            raise ValueError(f'SHA256 value"{indicator_value}" is invalid')
+
+    if indicator_type == 'urls':
```

Comment:
```suggestion
    elif indicator_type == 'urls':
```

please change the rest as well